triggers/tests/BUILD.bazel

```diff
@@ -0,0 +1,368 @@
+                "//libs-scala/caching",
+                "//libs-scala/contextualized-logging",
+                "//libs-scala/ledger-resources",
+                "//libs-scala/logging-entries",
+                "//libs-scala/ports",
+                "//libs-scala/resources",
+                "//libs-scala/scalatest-utils",
+                "//observability/tracing",
+                "//test-common",
+                "//triggers/runner:trigger-runner-lib",
+            ],
+        ),
+    ]
+    for lf_version in DAML_LF_VERSIONS
+    for suffix in [("-" + lf_version) if lf_version else ""]
+]
+
+da_scala_library(
+    name = "trigger-simulation-lib",
+    srcs = [
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/CatGenerators.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerRuleSimulationLib.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/LedgerProcess.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ReportingProcess.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/TriggerProcess.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerACSDiff.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerApiClient.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerExternalAction.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerRegistration.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/report/ACSReporting.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/report/MetricsReporting.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/wrapper/TriggerFilter.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/wrapper/TriggerInitialize.scala",
+        "src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/wrapper/TriggerTimer.scala",
+    ],
+    scala_deps = [
+        "@maven//:com_typesafe_akka_akka_stream",
+        "@maven//:com_typesafe_akka_akka_actor_typed",
+        "@maven//:org_scalacheck_scalacheck",
+        "@maven//:org_scalactic_scalactic",
+        "@maven//:org_scalatest_scalatest_core",
+        "@maven//:org_scalatest_scalatest_wordspec",
+        "@maven//:org_scalaz_scalaz_core",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":test-utils",
+        "//canton/it-lib",
+        "//daml-lf/data",
+        "//daml-lf/interpreter",
+        "//daml-lf/language",
+        "//daml-lf/parser",
```

Comment:
where this is used ?

Reply:
This is used to define the function `unsafeSValueFromLf` in the `TriggerMultiProcessSimulation` abstract class.

Reply:
I see fair enough. 
Maybe a good idea to write such an example.

Reply:
Absolutely. FYI, this example is pulled from the draft PR https://github.com/digital-asset/daml/blob/63c09b82dc28a21b4176a7336482c8e452b4aaf7/triggers/simulations/scala/Contention.scala.

I've intentionally split out this change to try and simplify the review process. Apologies if this has caused issues.